pkg/util/kube/utils.go

```diff
@@ -97,15 +100,22 @@ func EnsureNamespaceExistsAndIsReady(namespace *corev1api.Namespace, client core
 
 	// err will be set if we timed out or encountered issues retrieving the namespace,
 	if err != nil {
+		if terminatingNamespace {
+			return false, nsCreated, errors.Wrapf(err, "timed out waiting for terminating namespace %s to disappear before restoring", namespace.Name)
+		}
 		return false, nsCreated, errors.Wrapf(err, "error getting namespace %s", namespace.Name)
 	}
 
 	// In the case the namespace already exists and isn't marked for deletion, assume it's ready for use.
 	if ready {
 		return true, nsCreated, nil
 	}
-
-	clusterNS, err := client.Create(context.TODO(), namespace, metav1.CreateOptions{})
+	namespaceToCreate := namespace.DeepCopy()
```

Comment:
https://github.com/vmware-tanzu/velero/blob/b2f1588d2e54c6a9cfd560845394ff94bc1513e4/pkg/backup/item_backupper.go#L162C5-L165

It seems there is no need to delete the status and the DeletionTimestamp from the namespace.
The namespace in the deletion state will not be included.

Reply:
Ok cool!. Thanks for the review, didn't know about this. Will remove this part then.

Hopefully no one has a use case for restoring an item already in deletion as new.

Reply:
addressed in 4e7fd6e96d28a2cb815e9c5ac63af47c05cc2883

Reply:
updated changelog in 0b36bb5e2

Reply:
undo utils_test 63fb95341e416e43ca86f2cddd5ddfbe67ea7eb7